client/internal/engine.go

```diff
@@ -1109,6 +1116,9 @@ func (e *Engine) protoRuleToFirewallRule(r *mgmProto.FirewallRule) firewall.Rule
 		protocol = firewall.ProtocolUDP
 	case "icmp":
 		protocol = firewall.ProtocolICMP
+	default:
+		log.Errorf("invalid protocol, skipping firewall rule: %q", r.Protocol)
+		return nil
```

Comment:
We should not apply partial update from management, so its better to attempt to convert all firewall rules form the update and if any of them are invalid abort the call to applyFirewallRules

Reply:
It is better to validate the rule in other places. That error in this method is critical and should not happen at all and mostly added by me for debugging reasons, because work is not finished yet in the logic layer.

Reply:
that's the point, if there is an error, the whole ACL update must not proceed and no changes should be applied

Reply:
Fixed in the engine, if the rule applies failed in the update batch, the update canceled already applied rules from this batch removed.

Reply:
Ok, with the ACL engine we should have this decoupled. Let merge and then work on it there.